data/meson.build

```diff
@@ -2,32 +2,42 @@
 # install the icons
 install_data(
     'icons/hicolor/scalable/apps/@0@.svg'.format(meson.project_name()),
-    install_dir: join_paths(get_option('datadir'), 'icons/hicolor/scalable/apps')
+    install_dir: get_option('datadir') / 'icons' / 'hicolor' / 'scalable' / 'apps'
 )
 install_data(
     'icons/hicolor/scalable/apps/@0@-symbolic.svg'.format(meson.project_name()),
-    install_dir: join_paths(get_option('datadir'), 'icons/hicolor/scalable/apps')
+    install_dir: get_option('datadir') / 'icons' / 'hicolor' / 'symbolic' / 'apps'
 )
 
+# Install the desktop file
+desktop_file = i18n.merge_file(
+    '@0@.desktop'.format(meson.project_name()),
+    output: '@BASENAME@',
+    input: 'pkg/desktop/@0@.desktop.in'.format(meson.project_name()),
+    po_dir: meson.source_root() / 'po',
+    type: 'desktop',
+    install: true,
+    install_dir: get_option('datadir') / 'applications'
+)
 
 # Validate desktop file
 desktop_file_validate = find_program('desktop-file-validate', required:false)
 if desktop_file_validate.found()
   test (
     'Validate desktop file',
     desktop_file_validate,
-    args: join_paths(meson.current_build_dir (), '@0@.desktop'.format(meson.project_name()))
+    args: desktop_file.full_path()
```

Comment:
If you need to use `full_path()` here, you should probably add `depends: desktop_file`. `full_path()` apparently loses target tracking information, so meson _could_ try to run this test before the file actually exists. It's not likely, but I'd be careful because I've run into it in other scenarios.

Reply:
You might also be able to pass `desktop_file` as args, though I'm not sure if `merge_file()` returns the right type of object for that.